dimagi/utils/name_to_url.py

```diff
@@ -1,4 +1,9 @@
 import re
 
-def name_to_url(name):
-    return re.sub(r'[^0-9a-z]+', '-', name.strip().lower())
+def name_to_url(name, default=""):
+    url = re.sub(r'[^0-9a-z]+', '-', name.strip().lower())
+    if re.search('^[0-9\-]*$', url) and default:
+        url = "{}-{}".format(default, url)
+    url = re.sub(r'^-+', '', url)
+    url = re.sub(r'-+$', '', url)
```

Comment:
Probably quicker to replace lines 7 & 8 with `url = url.strip('-')`:

    >>>  '---foo-bar---'.strip('-')
    'foo-bar'